desk/sur/hooks.hoon

```diff
@@ -23,62 +25,67 @@
       state=vase
       config=(map nest config)
   ==
-::  $hooks: collection of hooks, the order they should be run in, and
-::          any delayed hooks that need to be run
+::  $hooks: collection of hooks, the order they should be run in, hooks
+::          running on a schedule, and any hooks waiting to run
 ++  hooks
-  $:  hooks=(map id hook)
-      order=(map nest (list id))
-      crons=(map id (map origin cron))
-      delayed=(map delay-id [=origin delayed-hook])
+  $:  hooks=(map id-hook hook)
+      order=(map nest (list id-hook))
+      crons=(map id-hook cron)
+      waiting=(map id-wait [=origin waiting-hook])
   ==
 +$  origin  $@(~ nest)
-+$  delay-id  id
-+$  schedule  [next=@da repeat=@dr]
-+$  cron
-  $:  hook=id
++$  cron  (map origin job)
++$  job
+  $:  =id-hook
       =schedule
       =config
   ==
-::  $delayed-hook: metadata for when a delayed hook fires from the timer
-+$  delayed-hook
-  $:  id=delay-id
-      hook=id
++$  schedule  [next=@da repeat=@dr]
++$  id-wait  @uv
+::  $waiting-hook: metadata for when a waiting hook fires from the timer
++$  waiting-hook
+  $:  id=id-wait
+      hook=id-hook
       data=vase
       fires-at=time
   ==
-::
+::  $config: configuration data for a hook instance
 +$  config  (map @t *)
+::
+::  $action: what we can do with a hook
 +$  action
   $%  [%add name=@t src=@t]
-      [%edit =id name=(unit @t) src=(unit @t) meta=(unit data:m)]
-      [%del =id]
-      [%order =nest seq=(list id)]
-      [%config =id =nest =config]
-      [%wait =id =origin schedule=$@(@dr schedule) =config]
-      [%rest =id =origin]
+      [%edit id=id-hook name=(unit @t) src=(unit @t) meta=(unit data:m)]
+      [%del id=id-hook]
+      [%order =nest seq=(list id-hook)]
+      [%config id=id-hook =nest =config]
+      [%cron id=id-hook =origin schedule=$@(@dr schedule) =config]
+      [%rest id=id-hook =origin]
   ==
+::
+::  $response: the result of an action on a hook
 +$  response
```

Comment:
If we want to follow the convention already:
```suggestion
+$  r-hook
```

Reply:
will leave for later